packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/scheduleWalkthrough.ts

```diff
@@ -0,0 +1,352 @@
+import fs from 'fs-extra';
+import inquirer from 'inquirer';
+import path from 'path';
+import { FunctionParameters } from 'amplify-function-plugin-interface';
+inquirer.registerPrompt('datetime', require('inquirer-datepicker'));
+import { CronBuilder } from '../utils/cronBuilder';
+const categoryName = 'function';
+
+var Days = {
+  MON: '1',
+  TUE: '2',
+  WED: '3',
+  THU: '4',
+  FRI: '5',
+  SAT: '6',
+  SUN: '7',
+};
+
+export async function scheduleWalkthrough(context: any, params: Partial<FunctionParameters>): Promise<Partial<FunctionParameters>> {
+  const projectBackendDirPath = context.amplify.pathManager.getBackendDirPath();
+  const resourceDirPath = path.join(projectBackendDirPath, categoryName, params.resourceName);
+  const cfnFileName = `${params.resourceName}-cloudformation-template.json`;
+  const cfnFilePath = path.join(resourceDirPath, cfnFileName);
+  let scheduleParams: Partial<FunctionParameters> = params;
+  if (params.cloudwatchEnabled === undefined || params.cloudwatchEnabled === 'false') {
+    if (await context.amplify.confirmPrompt.run('Do you want this function to be invoked on a schedule?', false)) {
+      scheduleParams.cloudwatchEnabled = 'true';
+      try {
+        let cloudWatchRule = await cronServiceWalkthrough(context);
+        scheduleParams.cloudwatchRule = cloudWatchRule;
+        if (context.input.command === 'update') {
+          //append cloudwatch events to CFN File
+          const cfnContent = context.amplify.readJsonFile(cfnFilePath);
+          cfnContent.Resources.CloudWatchEvent = {
```

Comment:
I would put this CFN into helper functions so you don't have to see it when you're trying to read code logic

Reply:
done